dspace-api/src/main/java/org/dspace/qaevent/service/QAEventSecurityService.java

```diff
@@ -0,0 +1,55 @@
+/**
+ * The contents of this file are subject to the license and copyright
+ * detailed in the LICENSE and NOTICE files at the root of the source
+ * tree and available online at
+ *
+ * http://www.dspace.org/license/
+ */
+package org.dspace.qaevent.service;
+
+import java.util.Optional;
+
+import org.dspace.content.QAEvent;
+import org.dspace.core.Context;
+import org.dspace.eperson.EPerson;
+
+/**
+ * Interface to limit the visibility of {@link QAEvent} to specific users.
+ *
+ * @author Andrea Bollini (andrea.bollini at 4science.com)
+ *
+ */
+public interface QAEventSecurityService {
+
+    /**
+     * Check if the specified user can see a specific QASource
+     * @param context   the context
+     * @param user      the eperson to consider
+     * @param qaSource  the qaSource involved
+     * @return          <code>true</code> if the specified user can eventually see events in the QASource
+     */
+    boolean canSeeSource(Context context, EPerson user, String qaSource);
```

Comment:
@tdonohue I apologize for having missed this comment.
Where the method is called, only the name from the qasource is available, so it is not advisable to do it because you only have the name, and retrieving the entire object would involve unnecessary additional operations (worsening performance). A refactoring can be considered in the future if it is really necessary.